ocollection.h

```diff
@@ -0,0 +1,90 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+struct OVector{
+    int size;
+    int capacity;
+    int* array;
+};
+
+void initializeVector(struct OVector* vector) {
+    vector->size = 0;
+    vector->capacity = 5;
+    vector->array = malloc(vector->capacity * sizeof(int));
+}
+
+int* createArray( int capacity) {
+    int* array = malloc(capacity+sizeof(int));
+    return array;
+}
+
+void deleteArray(struct OVector* vector) {
+    free(vector->array);
+}
+
+void resizeArray(struct OVector* vector, int newCapacity) {
+    int* newArray = createArray(newCapacity);
+    for (int i = 0; i < vector->capacity; i++) {
+        newArray[i] = vector->array[i];
+    }
+    deleteArray(vector);
+    vector->array = newArray;
+    vector->capacity = newCapacity;
+}
+
+void push_back(struct OVector* vector, int element) {
+    if (vector->size == vector->capacity) {
+        resizeArray(vector, vector->capacity * 2);
+    }
+    vector->array[vector->size++] = element;
+}
+
+void push_front(struct OVector* vector, int element) {
+    if (vector->size == vector->capacity) {
+        resizeArray(vector, vector->capacity * 2);
+    }
+    for (int i = vector->size; i > 0; i--) {
+        vector->array[i] = vector->array[i - 1];
+    }
+    vector->array[0] = element;
+    vector->size++;
+}
+
+void pop_back(struct OVector* vector ) {
+    if(vector->size > 0){
+        vector->size--;
+    }
+
+}
+
+void pop_front(struct OVector* vector ) {
+    if(vector->size > 0){
+        for (int i = 0; i < vector->size; i++) {
+            vector->array[i] = vector->array[i + 1];
+        }
+        vector->size--;
+    }
+
+}
+
+void clear(struct OVector* vector) {
+    vector->size = 0;
+    resizeArray(vector, 5);
```

Comment:
would be better to free this array with `deleteArray` function

Reply:
I think it's better to leave _clear()_ like that because in the realization of directory of c++ _vector_  work just like that.
https://www.geeksforgeeks.org/vector-erase-and-clear-in-cpp/
instead, I will add method _deleteArray_ that will free our vector.

Reply:
ok, let's leave it as it is